src/derivkit/utils/numerics.py

```diff
@@ -56,3 +66,242 @@ def relative_error(a: np.ndarray, b: np.ndarray) -> float:
     b = np.asarray(b, dtype=float)
     denom = np.maximum(1.0, np.maximum(np.abs(a), np.abs(b)))
     return float(np.max(np.abs(a - b) / denom))
+
+
+def evaluate_logprior(
+    theta: ArrayLike,
+    logprior: Callable[[NDArray[np.floating]], float] | None,
+) -> float:
+    """Evaluates a log-prior with robust handling of invalid values.
+
+    If no prior is provided, this function represents an improper flat prior
+    and returns zero. If a prior is provided, its value is interpreted as a
+    log-density defined up to an additive constant. Any non-finite value
+    (e.g., ``-inf`` or ``nan``) is treated as zero prior probability.
+
+    Args:
+        theta: Parameter vector at which to evaluate the prior.
+        logprior: Callable returning the log-prior density, or ``None`` to
+            indicate an uninformative (flat) prior.
+
+    Returns:
+        The log-prior value at ``theta``, or ``-inf`` if the prior assigns
+        zero probability to that point.
+    """
+    if logprior is None:
+        return 0.0
+    v = float(logprior(np.asarray(theta)))
+    return v if np.isfinite(v) else -np.inf
+
+
+def in_bounds_checker(
```

Comment:
```suggestion
def is_in_bounds(
```
I feel this is slightly more ideomatic, as in this case you'd have a test like
```
if not is_in_bounds:
   # do something
```